server.go

```diff
@@ -3871,7 +3886,10 @@ func (s *server) peerConnected(conn net.Conn, connReq *connmgr.ConnReq,
 		GetAliases:             s.aliasMgr.GetAliases,
 		RequestAlias:           s.aliasMgr.RequestAlias,
 		AddLocalAlias:          s.aliasMgr.AddLocalAlias,
-		Quit:                   s.quit,
+		PeerDataStore: s.peerStorageProvider.NewPeerStorageDB(
+			pubKeyBytes,
+		),
```

Comment:
fyi - this is what is causing the panic in the itests. 

Even though this returns a nil instance of `PeerStorageDB`, because you are assigning it to an _interface_, the interface nil check will actually return false. For an example of a workaround, look a few lines above at how the tower client manager is initialised:

```
        // If we directly set the peer.Config TowerClient member to the
	// s.towerClientMgr then in the case that the s.towerClientMgr is nil,
	// the peer.Config's TowerClient member will not evaluate to nil even
	// though the underlying value is nil. To avoid this gotcha which can
	// cause a panic, we need to explicitly pass nil to the peer.Config's
	// TowerClient if needed.
	var towerClient wtclient.ClientManager
	if s.towerClientMgr != nil {
		towerClient = s.towerClientMgr
	}
```

Reply:
Thanks I was thinking of doing a similar check as watchtower serve and client on the config builder level.

https://github.com/lightningnetwork/lnd/blob/24e19579939b4e7a1f009a32d0d41c120ea6dfe7/config_builder.go#L1060-L1072
Do you think it would be a good idea to separate out the db initliazation logic from `NewPeerStorageDB` into a new function `initPeerStoageDB` and do something like the above?

Also I was curious about this:
https://github.com/lightningnetwork/lnd/blob/51ebc2052f3e6ef3a3edf48cc0d213fbe7faff4e/lncfg/db.go#L673-L689

Why is it that it is only the bbolt backend that checks if towerClient and towerserver backend is enabled before initializing the towerClient and towerServer DB?

As I was also thinking of adding a new arg, peerStorageEnabled to this function:
https://github.com/lightningnetwork/lnd/blob/51ebc2052f3e6ef3a3edf48cc0d213fbe7faff4e/lncfg/db.go#L258-L261

So that we can also conditionally open the DB pending if PeerStorage is enabled or not

Reply:
> Why is it that it is only the bbolt backend that checks if towerClient and towerserver backend is enabled before initializing the towerClient and towerServer DB?

cause with bbolt it is a whole new file. Whereas with the rest, it's just new tables in the existing DB.

Reply:
> arg, peerStorageEnabled to this function:

Yeah could do! Although I dont think creating the file is a huge overhead. So imo not a big deal to just initialise it even if it is disabled. Especially cause this conditional init is just for bbolt which we will hopefully phase out completely soon enough

Reply:
> Do you think it would be a good idea to separate out the db initliazation logic from NewPeerStorageProducer into a new function initPeerStoageDB and do something like the above?

Yeah could do! That would defs make things easier later on if a migration is required etc. It would basically involve just adding all that boiler plate etc.